ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/ForeignKeyJoinParamsFactory.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams;
+
+import io.confluent.ksql.name.ColumnName;
+import io.confluent.ksql.schema.ksql.Column;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.LogicalSchema.Builder;
+import java.util.Optional;
+
+public final class ForeignKeyJoinParamsFactory {
+
+  private ForeignKeyJoinParamsFactory() {
+  }
+
+  public static <KRightT> ForeignKeyJoinParams<KRightT> create(
+      final ColumnName leftJoinColumnName,
+      final LogicalSchema leftSchema,
+      final LogicalSchema rightSchema
+  ) {
+    return new ForeignKeyJoinParams<>(
+        createKeyExtractor(leftSchema, leftJoinColumnName),
+        new KsqlValueJoiner(leftSchema.value().size(), rightSchema.value().size(), 0),
+        createSchema(leftSchema, rightSchema)
+    );
+  }
+
+  public static LogicalSchema createSchema(
+      final LogicalSchema leftSchema,
+      final LogicalSchema rightSchema
+  ) {
+    final Builder builder = LogicalSchema.builder()
+        .keyColumns(leftSchema.key())
```

Comment:
It occurs to me now that these column names might be aliased, which we probably don't want. Can we verify? Can also defer to a later PR when we wire everything together but my current suspicion is that this will be a problem.

Reply:
Not 100% sure atm -- `leftSchema.key()`returns `Column` that only has a `ColumnName` but no "alias".

But yes, we need to keep aliases in mind... I think it might be best to defer to later.